### DISCUSSION TITLE
Trouble getting my trigger function to work

### DISCUSSION THREAD

**Discussion by filigreti** [supabase/supabase#21163] [Questions]
Hello everyone, Im relatively new to postgres but i have been trying to learn as much as possible. I created a trigger that is suppose to activate after a user signs up to the platform. i have a users table, a roles table, and a user-roles table. the plan is to make a new record to the user table as well as update the user_roles tables via the is i send as roles

this is what i am sending as the payload
{
    "email": "one@example.com",
    "password": "iamtired",
    "options": {
        "data": {
            "roles": [
                2,
                1
            ],
            "first_time_login": true,
            "active": true
        }
    }
}

this is the trigger function
`declare
role_id int;

begin
    insert INTO users (id, email, active, first_time_login)
    values (new.id, new.email, new.raw_user_meta_data->>'active', new.raw_user_meta_data->>'first_time_login');
    FOR role_id IN SELECT json_array_elements_text(new.raw_user_meta_data->'roles')loop
        insert INTO user_roles (user_id, role_id)
        VALUES (new.id, role_id);
    end loop;
end;`

i have not been successfull, and i get kicked out of the app cause once i get an error when creating a user my session gets closed

**Comment by GaryAustin1** [comment]
You should look into the Postgres logs of the dashboard for the error.
Your function needs to be security definer type.  Any you need use public.user and public.user_roles or set the search_path parameter in your function. Those are two common issues.

Most of the startup guides show this trigger.
https://supabase.com/docs/guides/auth/managing-user-data#advanced-techniques

**Reply by GaryAustin1** [reply]
Did you look in the logs?
I suspect at a minimum you have boolean columns active and first_time_login and by using `->>` in the json data you are trying to send text (`->>` means treat the json value as text).  You probably will need to cast those as `(new.raw_user_meta_data->>'first_time_login')::boolean`
